File: LeftRightSUmDifferences.cpp
#include <iostream>
#include <vector>
using namespace std;

class LeftRightSUmDifferences {
public:
    vector<int> leftRightDifference(vector<int>& nums) {
        vector<int> ans(nums.size());

        int rightSum = 0;
        for (int num : nums) {
            rightSum += num;
        }

        int leftSum = 0;
        for (int i = 0; i < nums.size(); i++) {
            rightSum -= nums[i];
            ans[i] = abs(leftSum - rightSum);
            leftSum += nums[i];
        }

        return ans;
    }
};

int main() {
    vector<int> nums = {10, 4, 8, 3};

    LeftRightSUmDifferences lrsd;
    vector<int> result = lrsd.leftRightDifference(nums);

    for (int x : result) {
        cout << x << " ";
    }

    return 0;
}

File: Python/LeftRightSUmDifferences.py
class LeftRightSUmDifferences(object):
    def leftRightDifference(self, nums):
        n = len(nums)
        ans = [0] * n

        right_sum = sum(nums)
        left_sum = 0

        for i in range(n):
            right_sum -= nums[i]
            ans[i] = abs(left_sum - right_sum)
            left_sum += nums[i]

        return ans

nums = [10, 4, 8, 3]

lrsd = LeftRightSUmDifferences()
print(lrsd.leftRightDifference(nums))